x/mongo/driver/uuid/uuid.go

```diff
@@ -18,10 +17,10 @@ import (
 type UUID [16]byte
 
 // random is a package-global pseudo-random number generator.
-var random = randutil.NewLockedRand(rand.NewSource(time.Now().UnixNano()))
```

Comment:
I'm a little confused on why `time.Now().UnixNano()` doesn't work here, even if it returns the same value n-many times over some window.  [Here](https://go.dev/play/p/eW2L3pd-ykQ) is a simplified version of what this would look like in one build of our driver.  The only way I can make sense of this is if we have multiple builds of our driver being used to process data; something like [this](https://go.dev/play/p/RhPJ2osJBT8) where the source of random numbers is getting re-initialized multiple times with the same value.  Is that what's happening?

Reply:
If I understand correctly, the hypothetical issue with using `time.Now()` to seed the pseudo-random number is if two _concurrent_ (or otherwise simultaneously-running) processes use `newUUID()`. Those two processes would receive the same UUID if they called `newUUID()` at the same time. Seeding based on the `crypto/rand` reader will remove the possibility of two processes seeding with the same timestamp.

Reply:
I read through the [google/uuid](https://github.com/google/uuid) package and this is pretty much exactly what they do, so I closed the comment.  It's interesting we maintain this ourselves, though.

Reply:
@benjirewis's explanation is accurate and is actually reliably reproducible in the new `TestGlobalSource` unit test if `newGlobalSource` uses `time.Now().UnixNano()` instead of `randutil.CryptoSeed()`.

As far as your Go Playground examples, the real scenario is probably closest to the 3rd example, except that the driver doesn't seed a new `rand.Source` on every call for a random UUID. However, your 3rd Playground example encountering a duplicate value does expose a really strange behavior of Go's "math/rand" pseudo-random number generator: it throws away half of the 64 bits of entropy in the seed (see the code [here](https://cs.opensource.google/go/go/+/refs/tags/go1.18:src/math/rand/rng.go;l=204))! I modified your example to print the seed values that caused the duplicate and the truncated seed value actually used in `rand.NewSource()` and was able to reproduce duplication on my workstation after between 15,000 - 100,000 iterations. Check it out [here](https://go.dev/play/p/WXHUUKJKVY4).

E.g. output:
```
panic: duplication occurred at iteration 110784 with seeds -6055926323604102694(int32 seed 2140008456) and 4605785139232365995(int32 seed 2140008456): [18 30 190 192 107 168 222 204 240 58 165 60 211 193 105 224]
```

I've created [GODRIVER-2361](https://jira.mongodb.org/browse/GODRIVER-2361) to investigate using the [golang.org/x/exp/rand](https://pkg.go.dev/golang.org/x/exp/rand) package instead of "math/rand", which should alleviate that limitation. For now, the `TestGlobalSource` unit test asserts that there is no value duplication from `uuid.New()` when concurrently initializing 1,000 package-global UUID generators. However, that test does fail reliably if you push the number to 100,000. That's OK for practical use cases (even if two processes use the same effective seed, they are unlikely to use the same session UUID at the same time), but it's something we should fix.